### PR TITLE
docs: add Ionic + Capacitor integration

### PR DESCRIPTION
**Pull request by eduardoRoth**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/analogjs/analog/blob/main/CONTRIBUTING.md#-commit-message-guidelines
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] Other... Please describe:

## Which package are you modifying?

- [ ] vite-plugin-angular
- [ ] vite-plugin-nitro
- [ ] astro-angular
- [X] create-analog
- [ ] router
- [ ] platform
- [ ] content
- [ ] nx-plugin
- [ ] trpc

## What is the current behavior?
N/A

## What is the new behavior?
This PR adds Ionic+Capacitor integration to Analog in the following forms:
1. Documentation on how to integrate Ionic+Capacitor into an existing Analog app.
2. New Ionic template that generates an Analog app with Ionic integration.
3. New Ionic Capacitor template that generates an Analog app with Ionic+Capacitor integration.

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

## What [gif](https://chrome.google.com/webstore/detail/gifs-for-github/dkgjnpbipbdaoaadbdhpiokaemhlphep) best describes this PR or how it makes you feel?
image

### CODE REVIEW

**Comment by brandonroberts** [comment]
Wow, thanks @eduardoRoth! I think we should start with only the docs guide and make an example repository that we can link to instead of adding more templates to `create-analog` for now.

File: packages/create-analog/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 <a name="0.0.1"></a>
 
+## 1.2.3 (2024-04-29)
```
Comment (brandonroberts) [inline packages/create-analog/CHANGELOG.md:3]: Revert this

File: packages/create-analog/package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "create-analog",
-  "version": "1.2.2",
+  "version": "1.2.3",
```
Comment (brandonroberts) [inline packages/create-analog/package.json:3]: Revert this, it gets versioned in CI

File: apps/docs-app/docs/integrations/ionic/index.md [hunk 3]
```diff
@@ -0,0 +1,325 @@
+---
+sidebar_position: 4
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Ionic Integration with Analog
+
+This tutorial will guide you through the process of integrating Ionic Framework within your Analog application so you can leverage the power of Ionic's iOS and Android components in your applications.
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/ionic/index.md:10]:
```suggestion
This tutorial guides you through the process of integrating Ionic Framework within your Analog application so you can leverage the power of Ionic's iOS and Android components in your applications.
```

File: apps/docs-app/docs/integrations/ionic/index.md [hunk 4]
```diff
@@ -0,0 +1,325 @@
...
+
+3. Update `capacitor.config.ts` `webDir` property to point to the dist folder of analog build
+
+```ts
+import type { CapacitorConfig } from '@capacitor/cli';
+
+const config: CapacitorConfig = {
+  appId: 'com.ionic.capacitor',
+  appName: 'ionic-capacitor',
+  webDir: 'dist/client',
```
Comment (brandonroberts) [inline apps/docs-app/docs/integrations/ionic/index.md:264]:
```suggestion
  webDir: 'dist/analog/public',
```

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Comment by eduardoRoth** [comment]
@brandonroberts , branch updated! 😄

**Review by brandonroberts — APPROVED** [review]

**Comment by brandonroberts** [comment]
Thanks @eduardoRoth! @allcontributors add @eduardoRoth for doc

**Pull request merged**